### ISSUE TITLE
No error is thrown when port is taken

### ISSUE THREAD

**Issue by Electroid** [oven-sh/bun#1443]
```js
import { serve } from "bun";

serve({
  port: 4040,
  fetch(request) {
    return new Response("Hi");
  }
});

serve({
  port: 4040,
  fetch(request) {
    return new Response("Hi (again)");
  }
});
```

I would expect an error to be thrown on the 2nd `serve`. There is also no error when the port is taken by a non-Bun process. Tested on MacOS Silicon, while trying to debug #247

**Comment by Jarred-Sumner** [comment]
This is more complicated than it seems

SO_REUSEPORT is a builtin way to do load balancing, but it only really works properly on Linux

We should probably disable SO_REUSEPORT when development mode is enabled

**Comment by Electroid** [comment]
I think it's a potential footgun to not have this throw, even though `SO_REUSEPORT` is sometimes useful in certain deployments. Maybe that's a flag you have to opt-in to?

```js
export default {
  port: 3000,
  portReuse: true,
  // ...
}
```

**Comment by Electroid** [comment]
This was fixed.

**Comment by calebeaires** [comment]
@Electroid, when you mentioned that "we will modify the default behavior to indicate an error in the event of a port conflict," are you implying that the reusePort feature is being considered for deprecation?

I have found the reusePort feature to be highly beneficial, as it serves as a viable alternative to PM2 cluster mode!

**Comment by khaled-0** [comment]
This serves as a built in load balancer for me. Should it?
Seems like an unintentional feature to me.

**Comment by Jarred-Sumner** [comment]
@calebeaires @khaled-0 we are not deprecating `reusePort: true`, do not worry.

Originally, `reusePort: true` did not exist, it was implicitly the default behavior (i.e. no option), which means we did not throw an error when the port was already in use. This is very confusing in development, but in production this feature is important. So we added `reusePort: true` to make this behavior explicit - to allow the production usecase without causing confusion in development.
